Basic/DyanmicPrefabs/Assets/Scripts/01_Connection Approval/ConnectionApproval.cs

```diff
@@ -0,0 +1,133 @@
+            // soon as the server is started
+            // for more complex use cases where the server must wait for all connected clients to load the same network
+            // prefab, see the other use cases inside this sample
+            m_NetworkManager.OnServerStarted += LoadAPrefab;
+        }
+
+        async void LoadAPrefab()
+        {
+            await DynamicPrefabLoadingUtilities.LoadDynamicPrefab(
+                new AddressableGUID() { Value = m_AssetReferenceGameObject.AssetGUID}, 
+                0);
+        }
+
+        public override void OnDestroy()
+        {
+            m_NetworkManager.ConnectionApprovalCallback -= ConnectionApprovalCallback;
+            m_NetworkManager.OnServerStarted -= LoadAPrefab;
+            DynamicPrefabLoadingUtilities.UnloadAndReleaseAllDynamicPrefabs();
+            base.OnDestroy();
+        }
+    
+        void ConnectionApprovalCallback(NetworkManager.ConnectionApprovalRequest request, NetworkManager.ConnectionApprovalResponse response)
+        {
+            Debug.Log($"Client {request.ClientNetworkId} is trying to connect ");
+            var connectionData = request.Payload;
+            var clientId = request.ClientNetworkId;
+            
+            if (clientId == m_NetworkManager.LocalClientId)
+            {
+                //allow the host to connect
+                Approve();
+                return;
+            }
+            
+            if (connectionData.Length > k_MaxConnectPayload)
+            {
+                // If connectionData is too big, deny immediately to avoid wasting time on the server. This is intended as
+                // a bit of light protection against DOS attacks that rely on sending silly big buffers of garbage.
+                ImmediateDeny();
+                return;
+            }
+    
+            if (DynamicPrefabLoadingUtilities.LoadedPrefabCount == 0)
+            {
+                //immediately approve the connection if we haven't loaded any prefabs yet
+                Approve();
+                return;
+            }
+    
+            var payload = System.Text.Encoding.UTF8.GetString(connectionData);
+            var connectionPayload = JsonUtility.FromJson<ConnectionPayload>(payload); // https://docs.unity3d.com/2020.2/Documentation/Manual/JSONSerialization.html
+    
+            int clientPrefabHash = connectionPayload.HashOfDynamicPrefabGUIDs;
+            int serverPrefabHash = DynamicPrefabLoadingUtilities.ServerPrefabHash;
+            
+            //if the client has the same prefabs as the server - approve the connection
+            if (clientPrefabHash == serverPrefabHash)
+            {
+                Approve();
+
+                DynamicPrefabLoadingUtilities.RecordThatClientHasLoadedAllPrefabs(clientId);
+                
+                return;
+            }
+            
+            // In order for clients to not just get disconnected with no feedback, the server needs to tell the client
```

Comment:
I think these are copy paste comments from boss room, might be worth it to review and make sure they all make sense

Reply:
I just took a look and besides the payload comments, they're all unique to this sample.

Reply:
my bad, nvm